### PR TITLE
Split into matplotlib and matplotlib-base

### LINKED ISSUE #2: Split into meta packages
It would be great to split this recipe into several meta-packages.
That way, matplotlib doesn't have to depend on tk / qt etc. in the same way that jupyter doesn't pull in all of the dependencies needed for the notebook.

### PR DESCRIPTION
**Pull request by marcelm**
This closes #2 by splitting matplotlib into a `matplotlib` and a `matplotlib-core` package:

* `matplotlib` becomes a metapackage that depends on `matplotlib-core` and `pyqt`
* `matplotlib-core` is the same as the previous matplotlib package, except that it does not depend on `pyqt`.

Due to the new backend fallback mechanism in Matplotlib 3.0.0, a user who choses to install `matplotlib-core` will get the TkAgg backend automatically.

Closes #2. Supercedes #157.

I have had little experience with multiple outputs and it took me a while to get it to work, so I’ll appreciate feedback. In particular, I am not sure whether it was necessary to rename `build.sh`. The problem was that it was called even when packaging the `matplotlib-core` subpackage, which failed because the script assumes the build dependencies are available.

Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mingwandroid** [comment]
Could we use `base` instead of `core`, I only ask as there's some precedent for that (e.g. `r-base`).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,58 +15,64 @@ source:
...
-    - matplotlib._png
-    - matplotlib._tri
-    - matplotlib.backends._backend_agg
-    - matplotlib.ft2font
-    - matplotlib.ttconv
-    - matplotlib.backends._tkagg  # [not win]
-    - mpl_toolkits
-    - pylab
+outputs:
+  - name: matplotlib-core
```
Comment (ocefpaf) [inline recipe/meta.yaml]: The first `name` above is ignored and we are using `matplotlib-split` just as a convention. That means you need a second output that is named `matplolib` that depends on `matplolib-core` and the optional dependencies.

Pinging @msarahan here who knows more about split packages.

Comment (msarahan) [reply recipe/meta.yaml]: Yes, that's correct.  See the numpy recipe for reference: https://github.com/AnacondaRecipes/numpy-feedstock/blob/master/recipe/meta.yaml

Comment (marcelm) [reply recipe/meta.yaml]: Hm, but running `conda build` locally definitely produced two packages, and they even worked as expected when I installed them into test environments. I used the [implicit metapackages feature](https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#implicit-metapackages). Or do I misunderstand something?

**Comment by marcelm** [comment]
I like `-base` also better and changed it.

File: recipe/meta.yaml (hunk 1 above)
Comment (msarahan) [reply recipe/meta.yaml]: If it works, you're fine.  That has been fragile in the past, but hopefully we have it tested well enough now.

**Comment by mingwandroid** [comment]
Thanks @marcelm.

**Comment by marcelm** [comment]
Here comes a friendly reminder that this PR is still open.

**Comment by ocefpaf** [comment]
To avoid naming future naming issues with `defaults` I would appreciate @msarahan and @jjhelmus approvals here. (I guess we alreay have @mingwandroid's :wink:)

**Comment by ocefpaf** [comment]
Thanks @jjhelmus!

**Comment by marcelm** [comment]
Yay, thanks so much!

**Comment by ocefpaf** [comment]
> Yay, thanks so much!

Thank you for the PR! This was "in the works" since #2 but b/c `matplotlib` is so fundamental to the scientific stack we moved at a glacial pace here. Thanks for unlocking it!

**Pull request merged**